src/metrics/enhanced-metrics.ts

```diff
@@ -49,12 +49,17 @@ export function getRuntimeTag(): string | null {
   return `runtime:${processVersionTagString}`;
 }
 
-export function getEnhancedMetricTags(context: Context): string[] {
-  let arnTags = [`functionname:${context.functionName}`];
-  if (context.invokedFunctionArn) {
-    arnTags = parseTagsFromARN(context.invokedFunctionArn, context.functionVersion);
+export function getEnhancedMetricTags(context?: Context): string[] {
+  let tags: string[] = [];
+  if (context) {
+    let arnTags = [`functionname:${context.functionName}`];
+    if (context.invokedFunctionArn) {
+      arnTags = parseTagsFromARN(context.invokedFunctionArn, context.functionVersion);
+    }
+    tags = [...arnTags, `memorysize:${context.memoryLimitInMB}`];
```

Comment:
Does js have an `append`-like function? so we don't have to do the extra allocation.